javatools/jarutil.py

```diff
@@ -85,7 +85,7 @@ def verify(certificate, jar_file, sf_name=None):
 
     # Step 0: get the "key alias", used also for naming of sig-related files.
     zip_file = ZipFile(jar_file)
-    sf_files = filter(file_matches_sigfile, zip_file.namelist())
```

Comment:
what do you have against filter?

Reply:
In Python 3, `filter()` returns an iterator. Here we need a list, so the call should have been wrapped in an additional `list()`, redundant in Python 2. Just a list comprehension looks cleaner.